packages/blobs/src/store.ts

```diff
@@ -105,11 +108,51 @@ export type WriteResult = {
   modified: boolean
 }
 
+function otel<This extends Store, Args extends any[], Return>(getAttributes?: (...args: Args) => Attributes) {
+  return function (method: (...args: Args) => Promise<Return>, context: ClassMethodDecoratorContext) {
+    const methodName = String(context.name)
+    const operationName = `blobs.${methodName}`
+    return async function (this: This, ...args: Args): Promise<Return> {
+      const storeName = this['name']
+      const tracer = await getTracer()
+
+      if (tracer) {
+        return tracer.withActiveSpan(operationName, async (span) => {
+          span.setAttribute('blobs.store', storeName)
+          if (getAttributes) {
+            for (const [name, value] of Object.entries(getAttributes(...args))) {
+              value && span.setAttribute(`blobs.${name}`, value)
+            }
+          }
+          try {
+            this['span'] = span
+            const result = await method.apply(this, args)
+            span.setStatus({ code: SpanStatusCode.OK })
+            return result
+          } catch (error) {
+            const message = (error as Error).message ?? 'problem'
+            span.setStatus({
+              code: SpanStatusCode.ERROR,
+              message,
+            })
+            span.recordException(error as Error)
+            throw error
+          } finally {
+            this['span'] = null
+          }
+        })
+      }
+      return await method.apply(this, args)
+    }
+  }
+}
+
 export type BlobResponseType = 'arrayBuffer' | 'blob' | 'json' | 'stream' | 'text'
 
 export class Store {
   private client: Client
   private name: string
+  private span: Span | null = null
```

Comment:
I know this currently unused, but generally you wouldn't want to store a span like that because single blobs store can have multiple concurrent operations going on and this wouldn't work for such scenarios. 

If you need to get some active span - https://opentelemetry.io/docs/languages/js/instrumentation/#get-the-current-span is the way to go (not sure if `@netlify/otel` expose it or not, but if needed could be added there). That uses `async_hooks` to propagate active context/span through concurrent async hell

Reply:
thanks, i will look into that api